### PR TITLE
Support expireTime in URIs

### PR DESCRIPTION
**Pull request by paullinator**
### CHANGELOG

none

### Dependencies

none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204124160023937

### CODE REVIEW

File: src/actions/ScanActions.tsx [hunk 1]
```diff
@@ -217,7 +217,10 @@ export function handleWalletUris(
         tokenId
       }
 
-      navigation.push('send2', { walletId: wallet.id, spendInfo })
+      // React navigation doesn't like passing non-serializable objects as params. Convert date to string first
+      // https://github.com/react-navigation/react-navigation/issues/7925
+      const isoExpireDate = parsedUri?.expireDate != null ? parsedUri.expireDate.toISOString() : undefined
```
Comment (swansontec) [inline src/actions/ScanActions.tsx:222]: You can do `parsedUri?.expireDate?.toISOString()`. Stringing is fine, per our discussion the other day - it's not a public API, so go ahead and make the data easy to work with.

File: src/components/scenes/SendScene2.tsx [hunk 2]
```diff
@@ -839,6 +860,23 @@ const SendComponent = (props: Props) => {
...
+      const expireS = Math.max(Math.round(timeoutSeconds - nowSeconds), 0)
+      setExpireSeconds(expireS)
+      if (expireS <= 0) {
+        setError(new Error(s.strings.send_address_expired_error_message))
+      }
+      timeoutHandler.current = setTimeout(() => setTick(tick + 1), 500)
+      return () => clearTimeout(timeoutHandler.current)
+    }
+  }, [expireDate, tick])
+
```
Comment (swansontec) [inline src/components/scenes/SendScene2.tsx:879]: I wonder if we can factor out a `CountdownTile` component, that handles the timer internally and presents an `onDone` handler for when we reach the end. That way, we don't re-render the huge SendScene2 component 2x per second. React will only need to diff the tile itself.

The SendScene2 would continue to provide the `onDone` logic, of course, but that only fires once after the 10 minutes are up.

Comment (paullinator) [reply src/components/scenes/SendScene2.tsx]: good idea. done!

File: src/actions/ScanActions.tsx (hunk 1 above)
Comment (paullinator) [reply src/actions/ScanActions.tsx]: done

**Review by swansontec — APPROVED** [review]
Oooo, this is niiiiice.

**Pull request merged**